test/CMakeLists.txt

```diff
@@ -33,7 +33,11 @@ if(MIGRAPHX_DISABLE_LARGE_BUFFER_TESTS)
     add_compile_definitions(MIGRAPHX_DISABLE_LARGE_BUFFER_TESTS)
 endif()
 
+add_library(register_targets STATIC register_target.cpp)
+target_link_libraries(register_targets PRIVATE migraphx)
```

Comment:
I think you need to link `migraphx_all_targets` here as well to get the macros defines such as `HAVE_GPU`, `HAVE_CPU`, etc.

Reply:
Done